src/components/autonomous/WaypointNav.js

```diff
@@ -0,0 +1,73 @@
+import { useDispatch, useSelector } from "react-redux";
+import { selectLatitude, selectLongitude, selectIsApproximate, selectIsGated, requestWaypointNav } from "../../store/waypointNavSlice";
+import { selectOpMode } from "../../store/opModeSlice";
+import "./WaypointNav.css";
+import { useEffect } from "react";
+
+function WaypointNav() {
+  const dispatch = useDispatch();
+
+  var latitudeGiven = useSelector(selectLatitude);
+  var longitudeGiven = useSelector(selectLongitude);
+  var isApproximate = useSelector(selectIsApproximate);
+  var isGated = useSelector(selectIsGated);
+  var opMode = useSelector(selectOpMode);
+
+  const handleClick = () => {
+    console.log("hey");
+    updateLatitude();
+    updateLongitude();
+    updateIsApproximate();
+    updateIsGate();
+
+    // Only send the event in autonomous mode
+    if (opMode === "autonomous") {
+        dispatch(requestWaypointNav({ latitude: latitudeGiven,
+                                      longitude: longitudeGiven,
+                                      approximate: isApproximate,
+                                      gated: isGated}));
```

Comment:
Can we improve the formatting here? If we don't have a linter, we should add one